src/java/org/apache/cassandra/cql3/statements/schema/CreateKeyspaceStatement.java

```diff
@@ -75,8 +85,16 @@ public Keyspaces apply(Keyspaces schema)
             throw ire("Unable to use given strategy class: LocalStrategy is reserved for internal use.");
 
         keyspace.params.validate(keyspaceName);
+        Keyspaces ksps = schema.withAddedOrUpdated(keyspace);
 
-        return schema.withAddedOrUpdated(keyspace);
+        if (ClientWarn.instance.getWarnings() != null)
```

Comment:
and here same pattern: `clientWanrings.addAll(ClientWarn.instance.getWarningsAfter(previousWarningsSize));`

Reply:
Will resolve depending on how we resolve the above. Resolving for now to keep this clear-ish